components/net/http_loader.rs

```diff
@@ -1588,8 +1569,8 @@ async fn http_network_or_cache_fetch(
 
         // Step 14.3 If request’s use-URL-credentials flag is unset or isAuthenticationFetch is true, then:
         if !http_request.use_url_credentials || authentication_fetch_flag {
-            // TODO(#33616, #27439): Prompt the user for username and password from the window
-
+            let credentials = prompt_user_for_credentials(&context.state.embedder_proxy);
+            if let Some(_credentials) = credentials {}
```

Comment:
not questioning whether I need to do this, but wondering where in the spec it says to do this?

Reply:
Oh, huh. The fetch spec says to.update the request url's username and password. However, from a user experience perspective, we would definitely want to make it easy for the user to avoid inputting their credentials on every request that is behind a basic auth gateway.

Reply:
So I've implemented but it still infinite loops. And I assume you mean the next request is this line https://github.com/servo/servo/blob/11424f90b3cb0fe5688fce68139744f919c5be21/components/net/http_loader.rs#L1604

Reply:
I'll push out my changes so you can see it better